### ISSUE TITLE
Precision of sum

### ISSUE THREAD

**Issue by denisrosset** [gem-library/gem#18]
Say that `p` is a vector of random normal coefficients; we then compute their sum using GEM and the Symbolic toolbox:
```
p = randn(1, 1000);
s1 = sum(sym(p))
s2 = sum(gem(p))
```
gives (for our random seed)
```
s1 =
-4860173154195190169235169/112811063382770762907648
s2 = 
  -43.082415930293089851
```
Then we compute manually:
```
s3 = gem('-4860173154195190169235169')/gem('112811063382770762907648');

>> s2-s3
ans = 
  1e-14 *
   3.1805802961654285064
```
which seems to be much less precise than the default GEM precision.

Note that changing the number of digits `sum(gem(p,300))` does not change the result.

Originally, the discrepancy was noted when using the INTLAB precise summation algorithms (such as `Sum2`), but it's been changed to the Symbolic Toolbox here.

**Comment by jdbancal** [comment]
This is indeed surprising at first, but somehow expected. The basic reason is that `gem` is built with human users in mind. More specifically, it translates the fact that numbers in double precision can only be precise up to some point... essentially somewhere between 15 and 16 digits (depending on their actual value). Formally, they belong to a discrete grid, which might correspond to what `sym` can describe as well. But staying on this grid has concrete side-effects when one does not want to be restricted to 15 digits of precision, like with `gem`.

Essentially, for any number which is not a power of two, converting it to high precision requires choosing the value of new bits, which were not specified originally, and setting these bits to 0 in basis 2 corresponds to inventing new decimal digits... As an example, consider the following non-power of 2:
```
>> a = 1/3;
>> aTxt = num2str(a, 30)
aTxt =
    '0.333333333333333314829616256247'
```
The decimal expansion given here corresponds to a binary expansion where binary zeros were added at the end of the number. Therefore, the appearance of `616` in the decimal expansion, for instance, is clearly just noise, even though it looks very much like meaningful data. It is then easy to imagine that converting and including double numbers into a high-precision computation in this manner can easily lead to a lot meaningless information (and erasure of actual high precision information by noise).

The solution to this, of course, is to construct the number as a fraction of integers:
```
>> b = 1/gem(3);
>> num2str(b,30)
ans =
    '0.333333333333333333333333333333'
```
In order to remind the user of this good practice, the choice was made to base `gem`'s conversion on the human experience, and to set the digits of the newly-converted number to 0 in the natural basis 10. Concretely, 15 meaningful decimal digits are taken from the double value, and the remaining ones are set to zero. The result of simple operations is then easier to understand:
```
>> gem(1/3)-1/gem(3)
ans = 
  1e-16 *
  -3.3333333333333333333
```
instead of
```
>> gem(aTxt)-1/gem(3)
ans = 
  1e-17 *
  -1.8503717077086333333
```
This also allows a clear anticipation of how many decimal digits are preserved through the conversion (always 15).

Long story short, the solution here would be to sample the initial data in high precision to start with, i.e.
```
p = gem.randn(1,1000);
s1 = sum(sym(double(p)));
s2 = sum(p);
```
Then
```
s2 = 
  -39.144589335766717165
```
is the correct sum, up to the gem precision.

**Comment by jdbancal** [comment]
Ok, first note: `sym` does not guarantee an exact conversion from double, unless the argument `'f'` is provided.

Second, a new set of functions `gem2` and `sgem2` are now provided in 1aa4c5c42f6a6bc4c0796c5a057a744463e512ea to perform the 'exact' conversion from double to high precision number, i.e. by adding zeros to the binary expansion of the number.
